server/middlewares/userValidations.js

```diff
@@ -0,0 +1,38 @@
+import Validator from 'validatorjs';
+
+const signupValidations = (request, response, next) => {
+  const {
+    username, firstName, lastName, email, password
+  } = request.body;
+
+  const data = {
+    firstName: firstName && typeof firstName === 'string' ? firstName.trim() : firstName,
+    lastName: lastName && typeof lastName === 'string' ? lastName.trim() : lastName,
+    email,
+    username: username && typeof username === 'string' ? username.trim() : username,
+    password: password && password.trim()
```

Comment:
I think passwords can have trailing white spaces and might not require trimming.